cluster/gce/windows/common.psm1

```diff
@@ -146,6 +146,307 @@ function MustDownload-File {
   }
 }
 
+# This compiles some C# code that can make syscalls, and pulls the
+# result into our powershell environment so we can make syscalls from this script.
+# We make syscalls directly, because whatever the powershell cmdlets do under the hood,
+# they can't seem to open the log files concurrently with writers.
+# See https://docs.microsoft.com/en-us/dotnet/framework/interop/marshaling-data-with-platform-invoke
+# for details on which unmanaged types map to managed types.
+$SyscallDefinitions = @'
+[DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+public static extern IntPtr CreateFileW(
+  String lpFileName,
+  UInt32 dwDesiredAccess,
+  UInt32 dwShareMode,
+  IntPtr lpSecurityAttributes,
+  UInt32 dwCreationDisposition,
+  UInt32 dwFlagsAndAttributes,
+  IntPtr hTemplateFile
+);
+
+[DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+public static extern bool SetFilePointer(
+  IntPtr hFile,
+  Int32  lDistanceToMove,
+  IntPtr lpDistanceToMoveHigh,
+  UInt32 dwMoveMethod
+);
+
+[DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+public static extern bool SetEndOfFile(
+  IntPtr hFile
+);
+
+[DllImport("kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+public static extern bool CloseHandle(
+  IntPtr hObject
+);
+'@
+$Kernel32 = Add-Type -MemberDefinition $SyscallDefinitions -Name 'Kernel32' -Namespace 'Win32' -PassThru
+
+# ideally we can just use setFilePointer and setEndOfFile to truncate
+
+function Close-Handle {
```

Comment:
Can you add a small comment explaining what this function does?

Same for other functions you've added.

Reply:
Will do.